pkg/apis/agones/v1/gameserver.go

```diff
@@ -293,7 +293,7 @@ type GameServerStatus struct {
 	// +optional
 	Players *PlayerStatus `json:"players"`
 	// (Alpha, SafeToEvict feature flag) Eviction specifies the eviction tolerance of the GameServer.
-	Eviction Eviction `json:"eviction,omitempty"`
+	Eviction *Eviction `json:"eviction,omitempty"`
```

Comment:
@zmerlynn just noting I made this a pointer to be consistent with all other Alpha/Beta properties added to a CRD.

(I can't find the best practice doc I originally referenced for this approach.)

Reply:
Hmm, I saw that the others were but actually didn't understand why. It makes sense if they're `nullable`, but this field doesn't need to be nullable, I don't think.

Reply:
Aha, found the reference I was looking for:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

Having it be a non pointer means that if the feature flag is off, there is still some kind of data stored in etcd (probably a lot of empty strings, since that's the default). Whereas if it's a pointer, there is nothing to serialise/deserialise.

Reply:
Acctually, this is an optional field no -- I mean, the end user doesn't _have_ to set it? So we should probably mark it as such.

Reply:
Yes, I was just about to point out that by those conventions, we should add the `+optional` as well. But the concept LGTM - will give this a more thorough review in a bit (catching up).

Reply:
All good - no rush. I'll add in the `+optional` here as well.

Reply:
I approved but would like to see this before it goes in. :)

Reply:
Done!